packages/config/src/ConfigAutoStore.ts

```diff
@@ -184,8 +185,10 @@ export class ConfigAutoStore {
 
             // If secure array at higher level includes this property, then property should be stored at higher level
             if (isSecureProp) {
-                const secureProfilePath = config.api.secure.secureInfoForProp(`${propProfilePath}.properties.${propName}`, true).path;
-                if (secureProfilePath.split(".").length < propProfilePath.split(".").length) {
+                const secureProfile = config.api.secure.secureInfoForProp(`${propProfilePath}.properties.${propName}`, true);
+                let secureProfilePath: string;
+                if (secureProfile != null) secureProfilePath = secureProfile.path;
+                if (secureProfilePath != null && secureProfilePath.split(".").length < propProfilePath.split(".").length) {
```

Comment:
If we are concerned about # of lines of code for coverage purposes, this change might be able to shrink a bit?
```
                const secureProfile = config.api.secure.secureInfoForProp(`${propProfilePath}.properties.${propName}`, true);
                if (secureProfile != null && secureProfile.path.split(".").length < propProfilePath.split(".").length) {
```